great_expectations/rule_based_profiler/config/base.py

```diff
@@ -0,0 +1,282 @@
+        return self._module_name
+
+
+class ExpectationConfigurationBuilderConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    class_name = fields.String(required=True)
+    module_name = fields.String(
+        required=False,
+        all_none=True,
+        missing="great_expectations.rule_based_profiler.expectation_configuration_builder",
+    )
+    expectation_type = fields.String(required=True)
+    mostly = fields.Float(required=False, allow_none=True)
+    meta = fields.Dict(required=False, allow_none=True)
+
+    @post_load
+    def make_config(
+        self, data: dict, **kwargs
+    ) -> ExpectationConfigurationBuilderConfig:
+        return ExpectationConfigurationBuilderConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        res = copy.deepcopy(data)
+        for k, v in data.items():
+            if v is None:
+                res.pop(k)
+                logger.info("Removed '%s' due to null value", k)
+        return res
+
+
+class RuleConfig(DictDot):
+    def __init__(
+        self,
+        name: str,
+        domain_builder: DomainBuilderConfig,
+        parameter_builders: List[ParameterBuilderConfig],
+        expectation_configuration_builders: List[ExpectationConfigurationBuilderConfig],
+        **kwargs,
+    ):
+        self.name = name
+        self.domain_builder = domain_builder
+        self.parameter_builders = parameter_builders
+        self.expectation_configuration_builders = expectation_configuration_builders
+        for k, v in kwargs.items():
+            setattr(self, k, v)
+            logger.info(
+                'Set "self.%s = %s" during RuleConfig instantiation',
+                k,
+                v,
+            )
+
+
+class RuleConfigSchema(Schema):
+    class Meta:
+        unknown = INCLUDE
+
+    name = fields.String(required=True)
+    domain_builder = fields.Nested(DomainBuilderConfigSchema, required=True)
+    parameter_builders = fields.List(
+        cls_or_instance=fields.Nested(ParameterBuilderConfigSchema, required=True),
+        required=True,
+    )
+    expectation_configuration_builders = fields.List(
+        cls_or_instance=fields.Nested(
+            ExpectationConfigurationBuilderConfigSchema, required=True
+        ),
+        required=True,
+    )
+
+    @post_load
+    def make_config(self, data: dict, **kwargs) -> RuleConfig:
+        return RuleConfig(**data)
+
+    @post_dump
+    def remove_nulls(self, data: dict, **kwargs) -> dict:
+        res = copy.deepcopy(data)
+        for k, v in data.items():
+            if v is None:
+                res.pop(k)
+                logger.info("Removed '%s' due to null value", k)
+        return res
+
+
+class RuleBasedProfilerConfig(BaseYamlConfig):
+    def __init__(
+        self,
+        name: str,
+        config_version: float,
+        rules: Dict[str, RuleConfig],
+        variables: Optional[Dict[str, Any]] = None,
```

Comment:
can we do better than Any?

Reply:
Per the spec, it is an association between a string and one of: string, numeric, list, or dictionary. I don't think we can? @alexsherstinsky thoughts here?